subsys/mgmt/mcumgr/transport/src/smp_udp.c

```diff
@@ -167,57 +262,45 @@ static void smp_udp_receive_thread(void *p1, void *p2, void *p3)
 
 			smp_rx_req(&conf->smp_transport, nb);
 		} else if (len < 0) {
-			LOG_ERR("recvfrom error (%s): %i", conf->proto, errno);
+			LOG_ERR("recvfrom error (%s): %i, %d", smp_udp_proto_to_name(conf->proto),
+				errno, len);
 		}
 	}
 }
 
-static int smp_udp_init(void)
+static void smp_udp_net_event_handler(struct net_mgmt_event_callback *cb, uint32_t mgmt_event,
+				      struct net_if *iface)
 {
-	int rc;
+	ARG_UNUSED(cb);
+	ARG_UNUSED(iface);
 
+	if (mgmt_event == NET_EVENT_L4_CONNECTED) {
+		LOG_INF("Network connected");
 #ifdef CONFIG_MCUMGR_TRANSPORT_UDP_IPV4
-	configs.ipv4.smp_transport.functions.output = smp_udp4_tx;
-	configs.ipv4.smp_transport.functions.get_mtu = smp_udp_get_mtu;
-	configs.ipv4.smp_transport.functions.ud_copy = smp_udp_ud_copy;
-
-	rc = smp_transport_init(&configs.ipv4.smp_transport);
+		if (IS_THREAD_RUNNING(configs.ipv4.thread)) {
+			k_sem_give(&configs.ipv4.network_ready_sem);
+		}
 #endif
 
 #ifdef CONFIG_MCUMGR_TRANSPORT_UDP_IPV6
-	configs.ipv6.smp_transport.functions.output = smp_udp6_tx;
-	configs.ipv6.smp_transport.functions.get_mtu = smp_udp_get_mtu;
-	configs.ipv6.smp_transport.functions.ud_copy = smp_udp_ud_copy;
-
-	rc = smp_transport_init(&configs.ipv6.smp_transport);
+		if (IS_THREAD_RUNNING(configs.ipv6.thread)) {
+			k_sem_give(&configs.ipv6.network_ready_sem);
+		}
+#endif
+	} else if (mgmt_event == NET_EVENT_L4_DISCONNECTED) {
+		LOG_INF("Network disconnected");
+#ifdef CONFIG_MCUMGR_TRANSPORT_UDP_IPV4
+		if (IS_THREAD_RUNNING(configs.ipv4.thread)) {
+			k_sem_reset(&configs.ipv4.network_ready_sem);
```

Comment:
What is the reason for resetting the semaphores, since the threads are left running?
Would this be better moved inside `smp_udp_close()`?

Reply:
Was thinking of if the interface comes up then goes down close together but you are right it doesn't make sense, will remove

Reply:
Fixed